### DISCUSSION TITLE
how to use tsrange in colum helpers set ?

### DISCUSSION THREAD

**Discussion by pullmann4rent** [vitaly-t/pg-promise#914] [General]
Hello,

I want to use this:

`INSERT INTO booking (time_range) VALUES (tsrange($1, $2, '()')), [new Date(), new Date()]`

But how I use it with column helpers like this:

```
    const cs = new pgp.helpers.ColumnSet(
      ['user_id', 'tenant_id', 'location_id', 'start_time', 'end_time', 'time_range', 'status', 'info', 'created_by'], { table: 'workers_plan' });
  
    const cs_values: CreateWorkersPlan[] = plan.map((el: CreateWorkersPlan) => ({
      user_id: el.user_id,
      tenant_id,
      location_id: el.location_id,
      start_time: el.start_time,
      end_time: el.end_time,
      status: el.status,
      info: el.info,
      created_by
    }));

    const queryPlanInsert = pgp.helpers.insert(cs_values, cs);

    await pg.none(queryPlanInsert);
```

**Comment by vitaly-t** [comment]
Your insert sample and your columnset definition look like two unrelated things.

It helps if you try to create a related columnset, and if you have problems getting that insert query, then specify which problems exactly. Otherwise, it is too broad a question.

**Reply by pullmann4rent** [reply]
I want to add tsrange in columnset. In tsrange you have to add two values and this '()' but Idk how to  add this.

I can make this but this throw an error:
 ```
  const cs_values: CreateWorkersPlan[] = plan.map((el: CreateWorkersPlan) => ({
      user_id: el.user_id,
      tenant_id,
      location_id: el.location_id,
      start_time: el.start_time,
      end_time: el.end_time,
      time_range: [el.start_time, el.end_time, '()']  // THIS
      status: el.status,
      info: el.info,
      created_by
    }));

```
so how can I add this `(time_range) VALUES (tsrange($1, $2, '()'))` with columnset?

€:
`const tsrange = (start, end) => ({rawType: true, toPostgres: () => pgp.as.format('tsrange($1, $2)', [start, end])});`
I need this how can I implement this with columnset ?

**Reply by vitaly-t** [reply]
[Column](https://vitaly-t.github.io/pg-promise/helpers.Column.html) syntax has it all. What have you tried? If you have a specific problem using that syntax, then I can help.
